attributes/default.rb

```diff
@@ -30,10 +30,13 @@
 default['hops']['dir']                         = node['install']['dir'].empty? ? "/srv" : node['install']['dir']
 default['hops']['base_dir']                    = node['hops']['dir'] + "/hadoop"
 default['hops']['home']                        = node['hops']['dir'] + "/hadoop-" + node['hops']['version']
+default['hops']['data']['root_dir']            = "#{node['data']['dir']}/hops"
 
 default['hops']['sbin_dir']                    = node['hops']['base_dir'] + "/sbin"
 default['hops']['bin_dir']                     = node['hops']['base_dir'] + "/bin"
+default['hops']['data']['data_dir']            = "#{node['hops']['data']['root_dir']}/hopsdata"
 default['hops']['data_dir']                    = node['hops']['dir'] + "/hopsdata"
```

Comment:
Just to remove any disambiguation, `hops/data/data_dir` and `hops_data_dir` are not the same directory. The latter is a symlink to the former. `hops/data/data_dir = /srv/hopsworks-data/hops/hopsdata` while `hops/data_dir = /srv/hops/hopsdata`

I'm not very eager changing the directory structure completely. The same pattern is followed on all other cookbooks and while in some you can change the directory structure in others you can't (directories are created by the service).

As for the naming, I don't have any strong opinion `hops/data_volume/data_dir` maybe? I don't know

Reply:
I'm not arguing for changing the structure, just make the difference between `hops/data/data_dir` and `hops/data_dir` clear.

`hops/data_volume/data_dir` is good I think. We should also group all the `data_volume` attributes together so that it's clear their purpose. 
After that we should only use the `hops/data_dir` symlink in the configuration.